model-optimizer/mo/utils/cli_parser.py

```diff
@@ -917,44 +917,33 @@ def parse_tuple_pairs(argv_values: str):
     if not argv_values:
         return res
 
-    data_str = argv_values
-    while True:
-        tuples_matches = re.findall(r'[(\[]([0-9., -]+)[)\]]', data_str, re.IGNORECASE)
-        if not tuples_matches :
-            raise Error(
-                "Mean/scale values should be in format: data(1,2,3),info(2,3,4)" +
-                " or just plain set of them without naming any inputs: (1,2,3),(2,3,4). " +
-                refer_to_faq_msg(101), argv_values)
-        tuple_value = tuples_matches[0]
-        matches = data_str.split(tuple_value)
-
-        input_name = matches[0][:-1]
-        if not input_name:
-            res = []
-            # check that other values are specified w/o names
-            words_reg = r'([a-zA-Z]+)'
-            for i in range(0, len(matches)):
-                if re.search(words_reg, matches[i]) is not None:
-                    # error - tuple with name is also specified
-                    raise Error(
-                        "Mean/scale values should either contain names of input layers: data(1,2,3),info(2,3,4)" +
-                        " or just plain set of them without naming any inputs: (1,2,3),(2,3,4)." +
-                        refer_to_faq_msg(101), argv_values)
-            for match in tuples_matches:
-                res.append(np.fromstring(match, dtype=float, sep=','))
-            break
-
-        res[input_name] = np.fromstring(tuple_value, dtype=float, sep=',')
-
-        parenthesis = matches[0][-1]
-        sibling = ')' if parenthesis == '(' else ']'
-        pair = '{}{}{}{}'.format(input_name, parenthesis, tuple_value, sibling)
-        idx_substr = data_str.index(pair)
-        data_str = data_str[idx_substr + len(pair) + 1:]
-
-        if not data_str:
-            break
-
+    matches = [m for m in re.finditer(r'[(\[]([0-9., -]+)[)\]]', argv_values, re.IGNORECASE)]
+
+    error_msg = 'Mean/scale values should consist of name and values specified in round or square brackets' \
+                'separated by comma, e.g. data(1,2,3),info[2,3,4],egg[255] or data(1,2,3). Or just plain set of ' \
+                'values without names: (1,2,3),(2,3,4) or [1,2,3],[2,3,4].' + refer_to_faq_msg(101)
+    if not matches:
+        raise Error(error_msg, argv_values)
+
+    name_start_idx = 0
+    name_was_present = False
+    for idx, match in enumerate(matches):
+        input_name = argv_values[name_start_idx:match.start(0)]
+        name_start_idx = match.end(0) + 1
+        tuple_value = np.fromstring(match.groups()[0], dtype=float, sep=',')
+
+        if idx != 0 and name_was_present ^ bool(input_name):
+            # if node name firstly was specified and then subsequently not or vice versa
+            # e.g. (255),input[127] or input(255),[127]
+            raise Error(error_msg, argv_values)
+
+        name_was_present = True if input_name else False
```

Comment:
`name_was_present = input_name != ""` ?

Reply:
Yes, this line does `name_was_present = input_name != ""`

Reply:
I recommend to change the code like I proposed. For me it is more intuitive rather then `name_was_present = True if input_name else False`. However, I don't insist.

Reply:
Agree. Firstly I also had written `input_name != ""` but then googled and most of others use `if input_name` so I changed to more conventional. But `input_name != ""` is more explicit. Changed to it.